### ISSUE TITLE
[Feature Request]: Update Grant Amount in Celestia's FeeGrant Module

### ISSUE THREAD

**Issue by a26nine** [celestiaorg/celestia-node#3916]
## Feature Request: Update Grant Amount in Celestia's FeeGrant Module

**Description:**

The [feeGrant module](https://docs.celestia.org/how-to-guides/feegrant-for-blobs) in celestia-node enables a Granter to cover the transaction fees of a Grantee's blob submission. This is facilitated by granting a fee allowance, which authorizes the Grantee to submit blobs on behalf of the Granter up to a specified threshold.

**Problem:**

Currently, there's no way to modify the "grant amount" once it's set. Attempting to update it using the same process results in the following error:

```
failed to execute message; message index: 0: fee allowance already exists: invalid request
```

This behavior is evident in [this transaction](https://celenium.io/tx/36bd5ab0246db250b6e6c83ac2f5ab2242687d1bc98e2e414563f41f433da280?tab=messages), which fails when attempting to increase the allowance for an existing grant.

**Proposed Solution:**

Implement a feature that allows updating an existing grant amount. This would improve flexibility by eliminating the need to revoke the previous grant and create a new one each time a threshold adjustment is necessary.

**Benefits:**

- Simplifies workflows for managing allowances
- Reduces redundant transactions related to revoking and re-granting allowances
- Enhances usability for users leveraging the feeGrant module

**Comment by jcstein** [comment]
I agree that this would be useful. I think this also may need a corresponding issue to be made in https://github.com/celestia-app
